Aircraft/JA37/Nasal/payload/missiles.nas

```diff
@@ -5034,11 +5034,9 @@ var AIM = {
 		me.itVector = me.myMath.eulerToCartesian3X(-(meHeading+steering_h_deg), mePitch+steering_e_deg, 0);
 		me.itVector = me.myMath.normalize(me.itVector);
 		me.itVector = me.myMath.product(s_fps, me.itVector);#velocity vector if doing that steering
-		me.grVector  = [0,0,dt*g_fps];#velocity vector due to fighting gravity
```

Comment:
The missile don't work like that.
If the steering is 0 it means it do not want to deviate from current attitude.
Does no mean its in a freefall.
The missile-code does not model alpha (or beta), so when its in a freefall it will dive its nose down and have downwards speed.
Ergo it needs to be 1G when level and steering is 0 as that means its going to keep it current heading/pitch.

Reply:
Unguided bombs have steering=0 (line 2137)
```
me.observing = "unguided";
me.track_signal_e = 0;
me.track_signal_h = 0;
me.printGuide("Unguided");
```
Inversely, when level flight is desired, the elevator steering is positive to compensate gravity:
```
level: func () {
	me.attitudePN = math.atan2(-(me.speed_down_fps+g_fps * me.dt), me.speed_horizontal_fps ) * R2D;
	me.gravComp = me.pitch - me.attitudePN;
	#printf("Gravity compensation %0.2f degs", me.gravComp);
	me.track_signal_e = me.gravComp * !me.free;
```

If that doesn't convince you, just test it (with `DEBUG_FLIGHT`).
In the old version, free falling bombs report 1G, missile in level flight (guidance `level`) report 2G.
With my version, they report 0 and 1G respectively.

Reply:
This is the code which causes free falling when steering is 0 (around line 2173)
```
me.speed_down_fps       = -math.sin(me.pitch * D2R) * me.new_speed_fps;
[...]
me.speed_down_fps      += g_fps * me.dt;
[...]
me.pitch = math.atan2(-me.speed_down_fps, me.speed_horizontal_fps ) * R2D;
```

Reply:
Okay, then there is a problem. But just removing gravity from the G force calculation is not the way to fix it.
For unguided it should be 0 steering. Plus gravity effect. It might have to do with G being calculated with old speed.

Reply:
If steering is 0, the old function is equivalent to returning `cos(pitch)`.
So the old function can not be correct if steering is 0 for free falling stuff.

Reply:
What I mean is it uses the new heading and pitch, but the old speed from last loop, not even new speed (which is calculated before gravity effect is added).

Reply:
I actually think my version makes sense physically.

G-load can be defined as the sum of forces on the aircraft _excluding gravity_. (Gravity is excluded because it is the only force which also applies to occupants of the aircraft, and not 'just to the airframe').
Here we're only interested about G-load on the vertical and lateral axes, so we can forget about thrust and drag.
So the only force remaining is lift from the various control surfaces, which is somehow abstracted by the steering input.

Reply:
Alternatively, you could use old pitch / new pitch as inputs to compute G-load (and in that case add back gravity).

But I am certain that it is equivalent to my version.

Reply:
Oh, I am wrong that code was changed.
The G calc is now done before gravity is added in this loop.
It uses heading/pitch/speed from last loop, which should include 1G down though.
The G force calc should include gravity, as there is no alpha so gravity just means it pitch down a bit on top of steering. The code for that is just under guidance. My old code didn't do it proper, it now does.

I did find the issue though:
```javascript
# check if the missile overloaded with G force.
me.g = me.steering_speed_G(me.hdg, me.pitch, me.track_signal_e, me.track_signal_h, me.old_speed_fps, me.dt);
```

It uses the new steering command, speed from last loop, but also the new heading/pitch that already includes the new steering.

Reply:
I didnt see all your replies when I posted, will read them now.